.gitbook/0-about-panacea/1-data-exchange-protocol/2-How-DEP-works.md

```diff
@@ -0,0 +1,69 @@
+# How DEP Works
+
+In this guide, you can see how we solved the problem described on [DEP and the problems it solves](./1-DEP-problems-it-solves.md);
+
+- [For data privacy](#for-data-privacy)
+- [For data validation](#for-data-validation)
+- [Atomic incentive](#atomic-incentive-distribution)
+- [Decentralization & Scalability](#decentralization--scalability)
+
+## DEP flow
+
+Before reading on, if you want to know the entire flow of a DEP, check out this [user-flow](../../3-protocol-devs/1-dep-specs/1-user-flow.md)
+
+## For data privacy
+
+### Data privacy between provider & oracle
+
+In the process of [Data Validation](../../3-protocol-devs/1-dep-specs/4-data-validation.md), data provider first sends data to oracle for data verification and receives a certificate.
+In this process, the provider encrypts the data with the public key of the oracle and its own private key.
+The oracle private key is sealed in the enclave environment, so the data cannot be decrypted by anyone except the oracle.
+
+### Data privacy in oracle
+
+In the process of [Data Validation](../../3-protocol-devs/1-dep-specs/4-data-validation.md), the oracle decrypts the encrypted data with oracle private key and obtains the original data.
+However, the oracle we defined must run in an enclave environment, and the enclave memory cannot be read or written outside the enclave.
+Therefore, it is impossible to leak the original data and data privacy is continuously protected.
+
+### Data privacy between provider & consumer
+
+After completing the [Data Validation](../../3-protocol-devs/1-dep-specs/4-data-validation.md), oracle encrypts the data and provides this encrypted to the consumer.
+t this time, a symmetric key based on the oracle private key is generated, so the data cannot be decrypted until the consumer also receives this symmetric key.
+After the [Incentive Distribution](../../3-protocol-devs/1-dep-specs/6-incentives.md) process is completed, the consumer can request the symmetric key from the oracle.
+The oracle checks whether the incentive distribution process is completed from the panacea blockchain, and provides the secret key encrypted based on the consumer's public key.
```

Comment:
Rather than the incentive distribution process, why not modify it to submit consent and receive incentives?

Reply:
Thanks, it looks good.